PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,880 @@
+        :returns: dict
+        """
+        masking_view = None
+        try:
+            masking_view = self.conn.provisioning.get_masking_view(
+                mv_name)
+        except exception.ResourceNotFoundException:
+            print_str = 'Masking view %s not found.'
+            self.smart_print(print_str, WARNING, mv_name)
+        return masking_view
+
+    def get_or_create_masking_view(
+            self, element_dict, portgroup, host, revert=False):
+        """Get or create masking view from component elements.
+
+        :param element_dict: element dictionary
+        :param portgroup: port group name
+        :param host: host name
+        :param test: is it a test case
+        :returns: dict
+        """
+        new_masking_view_details = self.get_masking_view(
+            element_dict['new_mv_name'])
+        if new_masking_view_details:
+            if self.validate_existing_masking_view(
+                    new_masking_view_details, portgroup, host,
+                    element_dict, revert):
+                print_str = (
+                    'The existing masking view %s will be used.')
+                self.smart_print(
+                    print_str, DEBUG, element_dict['new_mv_name'])
+                self.smart_print('\n', DEBUG)
+            else:
+                print_str = (
+                    'Something is wrong with the existing masking view %s.')
+                self.smart_print(
+                    print_str, WARNING, element_dict['new_mv_name'])
+        else:
+            print_str = 'Creating masking view %s.'
+            self.smart_print(print_str, DEBUG, element_dict['new_mv_name'])
+            self.get_or_create_elements(element_dict, revert)
+            new_masking_view_details = self.get_masking_view(
+                element_dict['new_mv_name'])
+        return new_masking_view_details
+
+    def move_vols_from_source_to_target(
+            self, device_ids, source_sg, target_sg, create_vol_f):
+        """Get or create masking view from component elements.
+
+        :param device_ids: list of device ids
+        :param source_sg: the source sg
+        :param target_sg: the target sg
+        :param create_vol_f: create volume flag
+        :returns: string
+        """
+        print_str = '\nMoving %d volume(s) from %s to %s.'
+        self.smart_print(
+            print_str, DEBUG, len(device_ids), source_sg, target_sg)
+        self.smart_print(
+            '\nPlease be patient, this may take several minutes...',
+            DEBUG)
+        # Create a small volume
+        if create_vol_f:
+            self.conn.provisioning.create_volume_from_sg_return_dev_id(
+                'last_vol', source_sg, '1')
+        # Move the volume from the old storage group to the
+        # new storage group
+        message = self.conn.provisioning.move_volumes_between_storage_groups(
+            device_ids, source_sg, target_sg,
+            force=False
+        )
```

Comment:
The `force `and trailing `) `could be moved up onto the second line of the method call